0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,948 @@
+
+<!-- test-ignore -->
+```typeql
+define
+
+membership sub relation, relates parent, relates member;
+collection-membership sub membership;
+```
+
+In the example above, the `collection-membership` relation type inherits the `parent` and `member` roles defined in 
+its parent type, `membership`.
+
+The names of the inherited roles can be overridden to distinguish between the roles inherited by a relation subtype 
+vs. the roles defined by its parent type. However, the relation type being subtyped must be abstract. 
+It is not possible to override roles inherited by concrete relation types.
+
+<!-- test-ignore -->
+```typeql
+define
+
+membership sub relation, abstract, relates parent, relates member;
+
+collection-membership sub membership, 
+relates parent-collection as parent, 
+relates collection-member as member;
+```
+
+In the example above, the `collection-membership` relation type subtypes the `membership` relation type, and overrides 
+the inherited `parent` role as `parent-collection` and the inherited `member` role as `collection-member`.
+
+<div class="note">
+[Note]
+The two examples above can be run back to back. The second one will update the `membership` relation type to make it 
+abstract, and the `collection-membership` type to override its inherited roles.
+</div>
+
+##### Complex example
+
+<!-- test-ignore -->
+```typeql
+define
+
+ownership-type sub attribute, value string;
+
+ownership sub relation, abstract, 
+relates owned, 
+relates owner;
+
+group-ownership sub ownership, 
+relates owned-group as owned, 
+relates group-owner as owner, 
+owns ownership-type;
+
+object-ownership sub ownership, 
+relates owned-object as owned, 
+relates object-owner as owner, 
+owns ownership-type;
+
+access sub relation, 
+relates accessed-object, 
+relates valid-action, 
+plays change-request:requested-change;
+
+change-request sub relation, 
+relates requesting-subject, 
+relates requested-subject, 
+relates requested-change;
+```
+
+The example above defines one attribute type and five relation types:
+
+* `ownership` — subtypes the base `relation` type, is abstract, and relates `owned` and `owner` roles.
+* `group-ownership` — subtypes `ownership` relation type, relates `owned-group` as `owned` and `group-owner` as `owner`.
+* `object-ownership` — subtypes `ownership` relation, relates `owned-object` as `owned` and `object-owner` as `owner`.
+* `access` — subtypes the `relation` base type, relates `accessed-object` (e.g., files) and `valid-actions` (e.g., 
+  read), plays the role of `requested-change` in `change-request` relation type.
+* `change-request` — subtypes the `relation` base type, relates `requesting-subject`, `requested-subject` and
+  `requested-change`.
+
+### Define rules
+
+#### Overview
+
+Rules are defined independently of any types.
+
+They are executed only as a part of get queries when the [inference](06-infer.md) option is **enabled**. The results of 
```

Comment:
"part of match* queries" ? 
I'm not sure whether we use get as an umbrella term for other read-only match queries (e.g., group-aggregates) or whether it's a specific type of match query which has a `get` statement.

Reply:
We don't have a match query in our external terminology. Only get query (even if its without `get` clause). That way it is easier for users to remember it. We have discussed it with @flyingsilverfin some time ago.

P.S. Our internal terminology might differ from the external one - translated for users. But I'd prefer we use the same term everywhere. @flyingsilverfin even suggested we can rename the queries in our code.